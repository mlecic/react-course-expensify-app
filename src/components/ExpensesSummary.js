import React, { Component } from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export class ExpensesSummary extends Component {
    render() {
        const { expensesCount, expensesTotal } = this.props;
        return (
            <div>
                <p>
                    { 
                        expensesCount > 1 ?
                        `Viewing ${expensesCount} expenses totalling ${numeral(expensesTotal / 100).format('$0,0.00')}` :
                        `Viewing ${expensesCount} expense totalling ${numeral(expensesTotal / 100).format('$0,0.00')}`
                    }
                 </p>
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    const expensesSelected = selectExpenses(state.expenses, state.filters);
    return {
        expensesCount: expensesSelected.length,
        expensesTotal: selectExpensesTotal(expensesSelected)
    }
}

export default connect(mapStateToProps)(ExpensesSummary);
