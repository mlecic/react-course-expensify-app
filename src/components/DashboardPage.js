import React from 'react';
import ExpensesSummary from './ExpensesSummary';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';

const DashboardPage = () => (
    <div>
        <ExpensesSummary />
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default DashboardPage;