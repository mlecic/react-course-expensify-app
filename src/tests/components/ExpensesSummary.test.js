import React from 'react';
import { shallow } from 'enzyme';
// import expenses from '../fixtures/expenses';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render summary for single expense', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={500} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render summary for multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={2} expensesTotal={1000} />);
    expect(wrapper).toMatchSnapshot();
});