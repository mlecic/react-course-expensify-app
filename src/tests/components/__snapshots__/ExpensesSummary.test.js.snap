// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render summary for multiple expenses 1`] = `
<div>
  <p>
    Viewing 2 expenses totalling $10.00
  </p>
</div>
`;

exports[`should render summary for single expense 1`] = `
<div>
  <p>
    Viewing 1 expense totalling $5.00
  </p>
</div>
`;
